src/js/tests/ffi.spec.ts

```diff
@@ -17,22 +17,44 @@ describe('bindings', () => {
       27, 28, 29, 30, 31, 32,
     ])
 
-    const tree_index_le = new Uint8Array([
+    const treeIndexLE = new Uint8Array([
       33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55,
       56, 57, 58, 59, 60, 61, 62, 63, 64,
     ])
 
-    // Reverse the tree_index array so it is in little endian form
-    tree_index_le.reverse()
+    // Reverse the tree index array so it is in little endian form
+    treeIndexLE.reverse()
 
-    const sub_index = 0
+    const subIndex = 0
 
-    const key = context.getTreeKey(address, tree_index_le, sub_index)
-    const key_hex = bytesToHex(key)
+    const key = context.getTreeKey(address, treeIndexLE, subIndex)
+    const keyHex = bytesToHex(key)
 
     const expected = '0x76a014d14e338c57342cda5187775c6b75e7f0ef292e81b176c7a5a700273700'
 
-    expect(key_hex).toBe(expected)
+    expect(keyHex).toBe(expected)
+  })
+
+  test('getTreeKeyJsMatchesRust', () => {
+    const address = new Uint8Array([
+      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
+      27, 28, 29, 30, 31, 32,
+    ])
+
+    const treeIndex = new Uint8Array([
+      33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55,
+      56, 57, 58, 59, 60, 61, 62, 63, 64,
+    ])
```

Comment:
I am confused, taking this from the EIP https://eips.ethereum.org/EIPS/eip-6800 ...

```python
def get_tree_key(address: Address32, tree_index: int, sub_index: int):
    # Asssumes VERKLE_NODE_WIDTH = 256
    return (
        pedersen_hash(address + tree_index.to_bytes(32, 'little'))[:31] +
        bytes([sub_index])
    )
    
def get_tree_key_for_version(address: Address32):
    return get_tree_key(address, 0, VERSION_LEAF_KEY)
```

... `tree_index` is simply `0`, while above it is such a complex byte value. 🤔 How is this coming together?